In [1]:
#Importing reqiured libraries
import selenium
import pandas as pd
from selenium import webdriver
import requests

In [2]:
import time
from selenium.common.exceptions import NoSuchElementException
import warnings
warnings.filterwarnings('ignore')

In [4]:
#Connecting to web driver
driver = webdriver.Chrome(r"C://chromedriver.exe")
driver.maximize_window()

In [12]:
driver.get('http://amazon.in/')
driver.implicitly_wait(5)

In [13]:
# Let's enter the details in the search column
p_search=input("Enter product you want to search: ")
sear= driver.find_element_by_id("twotabsearchtextbox")
sear.send_keys(p_search)
sea_but=driver.find_element_by_id("nav-search-submit-button")
sea_but.click()

Enter product you want to search: laptops


In [14]:
rating = []
review = []

In [15]:
#Searching for laptops
#Functions for getting products in all 2000 entries 
urls=[]
for k in range(17):
    for i in driver.find_elements_by_xpath("//div[@class='a-section a-spacing-medium']//h2"):
        urls.append(i.find_element_by_xpath(".//a").get_attribute('href'))
    try:
        next_page=driver.find_element_by_xpath("//div[@class='a-text-center']/ul/li[@class='a-last']/a").get_attribute('href')
        driver.get(next_page)
        driver.refresh()   
    except:
        break

#Taking every URL one after one
for i in urls:
    driver.get(i)
    urls.append(i)
    driver.implicitly_wait(5)
    
#Scraping the rating
    try:
        driver.find_element_by_xpath("//a[1][@id='acrCustomerReviewLink']").click()
    except NoSuchElementException:
        print("No rating")
        pass
    
#Scraping all the reviews
    try:
        driver.find_element_by_xpath("//a[@class='a-link-emphasis a-text-bold']").click()
    except NoSuchElementException:
        pass

    
#Scrapping the details
    S_page=1
    E_page=60
    for page in range(S_page,E_page+1):
        try:
            reviews=driver.find_elements_by_xpath("//div[@class='a-row a-spacing-small review-data']/span/span")
            for r in reviews:
                review.append(r.text.replace('\n',''))
        except NoSuchElementException:
            review.append("Not Available")
        try:
            Rating=driver.find_elements_by_xpath("//div[@class='a-section celwidget']/div[2]/a[1]")
            for i in Rating:
                ratings=i.get_attribute('title')
                rating.append(ratings[:3])
        except NoSuchElementException:
            rating.append("Not available")      
            
        #Printing the page scrapped
        print("Product review and rating  of page {} scraped ".format(page+1))
        
        #Looping for going to next page automatically
        try:
            next_page=driver.find_element_by_xpath("//div[@id='cm_cr-pagination_bar']/ul/li[2]/a")
            if next_page.text=='Next Page':
                next_page.click()
                time.sleep(2)
        except NoSuchElementException:
            pass


Product review and rating  of page 2 scraped 
Product review and rating  of page 3 scraped 
Product review and rating  of page 4 scraped 
Product review and rating  of page 5 scraped 
Product review and rating  of page 6 scraped 
Product review and rating  of page 7 scraped 
Product review and rating  of page 8 scraped 
Product review and rating  of page 9 scraped 
Product review and rating  of page 10 scraped 
Product review and rating  of page 11 scraped 
Product review and rating  of page 12 scraped 
Product review and rating  of page 13 scraped 
Product review and rating  of page 14 scraped 
Product review and rating  of page 15 scraped 
Product review and rating  of page 16 scraped 
Product review and rating  of page 17 scraped 
Product review and rating  of page 18 scraped 
Product review and rating  of page 19 scraped 
Product review and rating  of page 20 scraped 
Product review and rating  of page 21 scraped 
Product review and rating  of page 22 scraped 
Product review and ra

KeyboardInterrupt: 

In [17]:
print(len(rating))

22620


In [20]:
print(len(review))

22620


#### Here, we have collected info of 22620 rows of different laptop reviews and ratings.

In [26]:
#Saving in a dataframe
lapdf=pd.DataFrame({'Reviews: ':review[:22000],'Ratings: ':rating[:22000]})

In [27]:
lapdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22000 entries, 0 to 21999
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   Reviews:   22000 non-null  object
 1   Ratings:   22000 non-null  object
dtypes: object(2)
memory usage: 343.9+ KB


In [28]:
#Starting 5 rows of dataframe
lapdf.head()

Reviews:  Ratings: 
0  Nice Product - The large screen and decent pro...       5.0
1  Description said 15.6". Laptop arrived. Laptop...       1.0
2  Bulid of laptop is sturdyBettery is also goodD...       4.0
3  Got this delivered today, this laptop is a bea...       5.0
4  Nice Product - The large screen and decent pro...       5.0

In [29]:
#Last 5 rows of dataframe
lapdf.tail()

Reviews:  Ratings: 
21995  This laptop comes with ryzen 5000 series proce...       4.0
21996  the features this laptop serves for the price ...       5.0
21997  The invoice doesn't contain the serial number ...       1.0
21998   Check full review in U tube from KM's frank view       3.0
21999  With Intelligent Fan cooling setup, Cinebench ...       4.0

In [30]:
#Saving the data into a csv file
lapdf.to_csv('RatRew.csv')